canonical-web-and-design/maas-ui
# refactor(ui): add a component for handling the network tab tables/forms state

## Done

- Make a reusable component for handling the state of the tables and forms in the network tabs.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the network tab in machine details.
- You should see the interfaces and DHCP tables.
- Click "Add interface".
- You should see the form appear at the bottom of the table.
- Click one of the "Edit..." links in the action menu.
- The content should be replaced with the form.

## Fixes

Fixes: canonical-web-and-design/app-tribe#563.
